server/src/services/projectService.ts

```diff
@@ -597,6 +598,64 @@ const setScheduleData = async (schedulePayload: SetSchedulePayload) => {
   return statusPayload;
 }
 
+const editScheduleData = async (payload: SetSchedulePayload) => {
+  try {
+    const scheduleData: ScheduleDataMongoResponse = await ScheduleDataSchema.findOne({"projectID": payload.projectID});
+    const keyData: ScheduleKeys | null = scheduleData.scheduleKeys.get(payload.locationID) ?? null;
+    
+    if (keyData) {
+      const parts = keyData.key.split(" ");
+      const date = parts.slice(0, 3).join(' ');
+      const time = parts.slice(3).join(' ');
+      const validDate = date.replace(/(\d+)(st|nd|rd|th),/, '$1,');
+      const dateUnix = new Date(validDate).getTime()
+
+      const oldPayload = {
+        time: time,
+        date: date,
+        dateUnix: dateUnix,
+        projectID: payload.projectID,
+        locationID: payload.locationID,
+        noteMessage: payload.noteMessage,
+        noteName: payload.noteName,
+        notePriority: payload.notePriority,
+        duration: keyData.duration,
+      }
+
+      const responseDel = await projectService.deleteSchedule(payload.locationID, payload.projectID);
+      //If there are schedule conflicts restore old data and throw error
+      const conflict = await checkIfConflictsExists(payload);
```

Comment:
the `checkIfConflictsExists` looks like it just calls `identifyNumOfConflicts`, then returns true or false based on the returned values. Is it possible to directly call `identifyNumOfConflicts` instead?

Reply:
Good point! I got rid of the checkIfConflictExists. I was trying to make it reusable but I think it just creates more db calls